### DISCUSSION TITLE
Kernel.Clone doesn't clone FunctionInvocationFilters

### DISCUSSION THREAD

**Discussion by jimmystridh** [microsoft/semantic-kernel#6240] [General]
doing .Clone() on a Kernel doesn't bring along the FunctionInvocationFilters I had registered. That was unexpected to me considering plugin registrations comes along with the clone. Is this intentional or an oversight?

**Comment by dmytrostruk** [comment]
Hi @jimmystridh, I think it should work if you register filters with DI through `kernelBuilder.Services`. Which means that in the same way it should work when you add filters through kernel properties (e.g. `kernel.FunctionInvocationFilters`). I will create a PR with fix for that. Thanks for reporting this issue!

**Comment by jimmystridh** [comment]
Thanks for the quick attention!

**Comment by dmytrostruk** [comment]
@jimmystridh Here is PR: https://github.com/microsoft/semantic-kernel/pull/6241
